### PR TITLE
apply stricter mypy to all new files, and fix a few tidbits

### PR DESCRIPTION
**Pull request by altendky**
This enforces stricter, though not actually all that strict, hinting policies on all new files while leaving existing files exempt.  It also adjusts a couple files that I figured were easier to adjust than to try to handle otherwise.  Note that this will apply to all existing PRs once merged and those that introduce new files will find those files bound by the stricter checking.

Draft for:
- [x] Discussion of type hint policies
- [x] Waiting until January to discuss and deploy
- [x] Standalone wallet being merged into `main`
- [ ] Finish discussion of proper `clvm_rs.run_generator2()` return hint

### CODE REVIEW

File: chia/util/make_test_constants.py [hunk 1]
```diff
@@ -1,6 +1,10 @@
 from typing import Dict
 
-from chia.consensus.default_constants import DEFAULT_CONSTANTS, ConsensusConstants
+import chia.consensus.default_constants
+
+
+DEFAULT_CONSTANTS = chia.consensus.default_constants.DEFAULT_CONSTANTS
+ConsensusConstants = chia.consensus.default_constants.ConsensusConstants
```
Comment (arvidn) [inline chia/util/make_test_constants.py:7]: what rule prefers this way of importing names?

Comment (altendky) [reply chia/util/make_test_constants.py:7]: Looks like this particular spot was me getting off on a wrong tangent...  sorry.  Undoing that bit, and tweaking a couple other.

The problem was at the other end of the import where we had `from .constants import ConsensusConstants` and then tried to import `ConsensusConstants` here.  The `no_implicit_reexport` is the issue there.  I'll switch the import to just get `ConsensusConstants` from the original place.

Thanks.

File: chia/util/clvm.py [hunk 2]
```diff
@@ -1 +0,0 @@
-from clvm.casts import int_from_bytes, int_to_bytes  # noqa
```
Comment (altendky) [inline chia/util/clvm.py:1]: @richardkiss, Arvid suggested I should check with you about this removal.  I searched and it seemed like more places imported directly from `clvm` than from here so I went this route.  Thoughts?

The reason this is even a topic is the mypy setting related to not considering imports to be exports.  You would want to be warned if someone, for example, wrote `from chia.daemon.server import sys`.  I mean it's there, but...  This is being categorized the same way.  We can make mypy not complain if we want, but this didn't strike me as an obviously needed thing given our present usage patterns around the underlying functions.

Comment (richardkiss) [reply chia/util/clvm.py:1]: Sure, I'm okay with getting rid of this. Likely it was larger in the past, and it's outlived its usefulness.

**Review by richardkiss — DISMISSED** [review]
Approved, at least the part that mentioned me. I didn't look too closely at the other bits.

File: chia/types/blockchain_format/program.py [hunk 3]
```diff
@@ -246,7 +246,7 @@ def run_with_cost(self, max_cost: int, *args) -> Tuple[int, Program]:
 
     # returns an optional error code and an optional PySpendBundleConditions (from clvm_rs)
     # exactly one of those will hold a value
-    def run_as_generator(self, max_cost: int, flags: int, *args) -> Tuple[Optional[uint16], Optional[Any]]:
+    def run_as_generator(self, max_cost: int, flags: int, *args) -> Union[Tuple[None, Any], Tuple[uint16, None]]:
```
Comment (altendky) [inline chia/types/blockchain_format/program.py:249]: @arvidn, is this hint a more accurate representation of what `clvm_rs.run_generator2()` returns?  I think both the Python hints and whatever Rust calls them (I swear I'll learn Rust someday...) are accurate but overly broad compared to the two things that are actually returned.

https://github.com/Chia-Network/clvm_rs/blob/781a74438a9c4c1168b285b9fbbcb6ebd466e931/src/py/run_generator.rs#L145
```rust
) -> PyResult<(Option<ErrorCode>, Option<PySpendBundleConditions>)> {
```
https://github.com/Chia-Network/clvm_rs/blob/781a74438a9c4c1168b285b9fbbcb6ebd466e931/src/py/run_generator.rs#L170-L187
```rust
    match r {
        Ok((None, Some(spend_bundle_conds))) => {
            // everything was successful
            Ok((
                None,
                Some(convert_spend_bundle_conds(
                    py,
                    &allocator,
                    spend_bundle_conds,
                )),
            ))
        }
        Ok((error_code, _)) => {
            // a validation error occurred
            Ok((error_code, None))
        }
        Err(eval_err) => eval_err_to_pyresult(py, eval_err, allocator),
    }
```

Comment (arvidn) [reply chia/types/blockchain_format/program.py:249]: I suppose so. the `Any` was really just me being lazy trying to find out how to refer to the rust type.

**Review by arvidn — DISMISSED** [review]

**Review by jack60612 — DISMISSED** [review]
lots of changes but looks good

**Comment by altendky** [comment]
I'm breaking this out into a few small pieces (see mention links above) and then plan to make a new PR that's tidier to do the core mypy strictness enhancement.

**Comment by altendky** [comment]
superceded by https://github.com/Chia-Network/chia-blockchain/pull/10133

**Pull request closed without merging**